api/src/OpenTelemetry/Internal/Logs/Types.hs

```diff
@@ -19,20 +23,114 @@ module OpenTelemetry.Internal.Logs.Types (
   toShortName,
 ) where
 
+import Control.Concurrent (MVar, newMVar, withMVar)
 import Control.Concurrent.Async
 import Data.Function (on)
+import Data.HashMap.Strict (HashMap)
 import qualified Data.HashMap.Strict as H
 import Data.IORef (IORef, atomicModifyIORef, modifyIORef, newIORef, readIORef)
 import Data.Text (Text)
 import Data.Vector (Vector)
 import OpenTelemetry.Common (Timestamp, TraceFlags)
 import OpenTelemetry.Context.Types (Context)
-import OpenTelemetry.Internal.Common.Types (InstrumentationLibrary, ShutdownResult)
+import OpenTelemetry.Internal.Common.Types (ExportResult, InstrumentationLibrary, ShutdownResult)
 import OpenTelemetry.Internal.Trace.Id (SpanId, TraceId)
 import OpenTelemetry.LogAttributes
 import OpenTelemetry.Resource (MaterializedResources)
 
 
+{- | @LogRecordExporter@ defines the interface that protocol-specific exporters must implement so that they can be plugged into OpenTelemetry SDK and support sending of telemetry data.
+
+The goal of the interface is to minimize burden of implementation for protocol-dependent telemetry exporters. The protocol exporter is expected to be primarily a simple telemetry data encoder and transmitter.
+-}
```

Comment:
Is this documentation in the right place now that this is `LogRecordExporterInternal`?

Reply:
https://hackage.haskell.org/package/persistent-2.14.6.2/docs/Database-Persist-SqlBackend.html#v:mkSqlBackend